.github/workflows/km-ci-workflow.yaml

```diff
@@ -161,10 +150,95 @@ jobs:
           name: km_release
           path: /tmp/release
 
+      - name: capture submodules
+        run: git submodule status > /tmp/km_submodule_status
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: km_submodule_status
+          path: /tmp/km_submodule_status
+
+
+  # Builds a static linked, stripped binary for krun/crun
+  # This uses the NIX virtual machine, just like the crun CI.
+  # NIX takes a long time, so we build the static krun binary in
+  # parallel with the rest of KM build and test. We just the 
+  krun-static-build:
+    name: Build static KRUN binary
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - uses: actions/cache@v2
+        with:
+          path: .cache
+          key: nix-v1-${{ hashFiles('container-runtime/crun/nix/nixpkgs.json') }}
+
+      - run: |
+          set -x
+          sudo find / -name crun -type f
+          CRUN_DIR=$(pwd)/container-runtime/crun
+          if [[ -z $(ls -A /nix) ]]; then sudo docker run --rm --privileged -v /:/mnt nixos/nix cp -rfT /nix /mnt/nix; fi
```

Comment:
What;s the point of the `if` here? I assume the environment is the same every time, the directory is either there or not.

Reply:
This was lifted directly from the CRUN CI script. Perhaps a comment to that effect would be helpful.

Reply:
OK. I'm thinking krun is better off being separate tree with build artifact, and this flow should just pull that and use whenever krun is used. I know you guys decided not to do that, not really sure why. 

Either way it is good as is for now.